### ISSUE TITLE
Forecast schedule recovery

### ISSUE THREAD

**Issue by mileserickson** [anstosa/ferry.fyi#22]
![F08D7C73-1309-41EC-AC9C-115DBCBA2E42](https://user-images.githubusercontent.com/68300/62094150-f039b980-b230-11e9-9299-b5d5fcf8556e.png)

Mukilteo-Clinton ferries typically stay on time in the AM (uni-directional peak volumes eastbound) but get progressively later throughout the afternoon (bi-directional peak volumes). They then gradually return to on-time operation in the late evening. Currently late ferries are predicted to stay late all night.

**Comment by anstosa** [comment]
Yeah the delay number could be more sophisticated. Here's my proposal for a first step for building a predictive model here:

* Create the model on a nightly basis by pulling all sailing for a given weekly slot for the last 1-2 months (e.g. Saturday at 10:30am or Sunday at 9:30pm)
* Calculate the average delta in departure delays between the previous slot and the current one (e.g. +11mins or -4mins)
* Forecast delays by adding the current actual delay for the vessel to average delay delta for that weekly sailing slot

**Comment by mileserickson** [comment]
I'd add: a first principles approach might be based on something like:
(1) It takes `crossing_time` to load pedestrians, untie, cross, and dock, and unload pedestrians.
(2) Before we can leave on our next trip, we need to unload `n_unload` cars at a rate of `unload_rate` per minute.
(3) We also need to load `min(capacity, n_waiting)` cars at a rate of `load_rate` per minute.

So the forecast additional delay and/or schedule recovery rate can be directly tied to the forecast vehicle loads.

**Comment by anstosa** [comment]
Added a model for projecting schedule recovery
